component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/voiceotp/OneTimePasswordUtils.java

```diff
@@ -0,0 +1,247 @@
+     *                                  algorithms available.
+     * @throws InvalidKeyException      The secret provided was not a valid HMAC-SHA-256 key.
+     */
+    public static String generateAlphaNumericOTP(byte[] secret, long movingFactor,
+                                                 int codeDigits, boolean addChecksum,
+                                                 int truncationOffset)
+            throws NoSuchAlgorithmException, InvalidKeyException {
+
+        // Put movingFactor value into text byte array.
+        String result = null;
+        int digits = addChecksum ? (codeDigits + 1) : codeDigits;
+        byte[] text = new byte[8];
+        for (int i = text.length - 1; i >= 0; i--) {
+            text[i] = (byte) (movingFactor & 0xff);
+            movingFactor >>= 8;
+        }
+        // Compute hmac hash.
+        byte[] hash = hmacShaGenerate(secret, text);
+        // Put selected bytes into result int.
+        int offset = hash[hash.length - 1] & 0xf;
+        if ((0 <= truncationOffset) && (truncationOffset < (hash.length - 8))) {
+            offset = truncationOffset;
+        }
+        int firstBinary = ((hash[offset] & 0x7f) << 24) | ((hash[offset + 1] & 0xff) << 16)
+                | ((hash[offset + 2] & 0xff) << 8) | ((hash[offset + 3] & 0xff));
+        int secondBinary = ((hash[offset + 4] & 0x7f) << 24) | ((hash[offset + 5] & 0xff) << 16)
+                | ((hash[offset + 6] & 0xff) << 8) | ((hash[offset + 7] & 0xff));
+        result = Integer.toString(firstBinary, 36).concat(Integer.toString(secondBinary, 36)).toUpperCase();
+        while (result.length() < digits) {
+            result = "A" + result;
+        }
+        result = result.substring(result.length() - digits, result.length());
+        return result;
+    }
+
+    /**
+     * Generate the token.
+     *
+     * @param key                       The key
+     * @param base                      The base
+     * @param digits                    The number of digits
+     * @param isEnableAlphanumericToken A flag that indicates the token is alphanumeric or not
+     * @return the generated token
+     */
+    public String generateToken(String key, String base,
+                                int digits, boolean isEnableAlphanumericToken) throws AuthenticationFailedException {
+
+        int truncOffset = 0;
+        try {
+            if (isEnableAlphanumericToken) {
+                return generateAlphaNumericOTP(key.getBytes(), Long.parseLong(base), digits, false, truncOffset);
+            } else {
+                return generateOTP(key.getBytes(), Long.parseLong(base), digits, false, truncOffset);
+            }
+        } catch (NoSuchAlgorithmException e) {
+            throw new AuthenticationFailedException(" Unable to find the SHA256 Algorithm to generate OTP ", e);
+        } catch (InvalidKeyException e) {
+            throw new AuthenticationFailedException(" Unable to find the secret key ", e);
```

Comment:
```suggestion
            throw new AuthenticationFailedException("Unable to find the secret key.", e);
```

Adding more context to the log would be beneficial when troubleshooting